### DISCUSSION TITLE
environment variables

### DISCUSSION THREAD

**Discussion by rdilipk** [redpanda-data/connect#1174] [Q&A] [answered]
What's the correct way to read values from environment variables and construct a DSN at runtime for a sql_insert output processor? Or must some external bootstrapper-like program create the benthos config.yaml with these values before benthos can be started? IOW, I am wondering if this can be done from inside the config yaml itself.

**Comment by mihaitodor** [comment, marked as answer]
I think GitHub must've skipped some notification emails... If we're slow to reply here, please reach out via the [other means](https://www.benthos.dev/community).

Regarding your question, please look at https://www.benthos.dev/docs/configuration/interpolation/. If you configure `dsn: ${FOO}`, then Benthos will attempt to set the `dsn` to the contents of the `FOO` environment variable. You can even specify a default value like so: `dsn: ${FOO:foouser:foopassword@tcp(localhost:3306)/foodb}` if you don't want to get an empty string when `FOO` isn't set.

**Reply by rdilipk** [reply]
I figured this one out myself. I completely missed the alternate contact mechanisms. Will keep them in mind next time. Thanks!
